### ISSUE TITLE
react tutorial hello world

### ISSUE THREAD

**Issue by notewrangler** [nodeschool/discussions#1160]
Using Ubuntu 14.04. 

Navigated to the directory where the react app was going to go (maybe that was a mistake)...

I followed these instructions first: 
     HELLO_REACT
     Exercise 1 of 11

     First things first, let's print Hello World !

    Start by installing the required modules. 
    Run the four commands below.

    $ npm install express

    $ npm install body-parser

    $ npm install express-react-views

    $ npm install node-jsx

    Next, create program.js and copy the code below into the file.

first installs threw an error, suggested I install as root.  
typed this command: sudo su - 
then the install proceeded.  Installed all 4 packages, though express-react-views had a warning: 

    npm WARN optional dep failed, continuing fsevents@0.3.6

install proceeded from there.  I noticed, in SublimeText, that a ton of directories were added to my repo folder.  Wasn't sure if "install" was supposed to put them there or in bin/ or usr/lib or whatever...

In my repo folder, I created the file "program.js" there, and copied the code from the exercise. created folder "views," created "index.jsx" in that folder, copied and pasted the correct code.
tried to run this: 

    $ node program.js

got error (same as one below).  Decided to try going back to the beginning after cd-ing to root directory 
($ cd / ) and $ sudo su - again. 
 
installed all packages again, this time from root directory.  

The module folders that installed themselves in my app repo directory (I didn't delete them) did not include anything named express-react-views. I DID find a folder with that name inside /usr/lib/node_modules/learnyoureact/node_modules.
 

Tried to run the program again. 

Still got the same error, this: 

    module.js:340
    throw err;
          ^
Error: Cannot find module 'express-react-views'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/ralph/workspace/udemy/react/program.js:7:23)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10) 

Don't know what's going on or if I did something incorrectly.  I'll wait to hear back from somebody. 
Thanks!

**Comment by martinheidegger** [comment]
ping @tako-black

**Comment by tako-black** [comment]
@notewrangler 
Would you run `npm install express-react-views` at the same directory as `program.js`?
Have you try?

**Comment by notewrangler** [comment]
@ tako-black...I did that first, before attempting installation from the root directory.  It created a node_modules directory in my react directory (the place where the program.js file sits).  

Is that how it is supposed to work?  Is all of that code intended to be installed in the working directory, or are these utilities intended to be installed system wide (or both)?  The tutorial doesn't specify. 

When I ran the installers from the root directory, an express-react_views folder was created in /usr/lib/node_modules/learnyoureact/node_modules.  

However, both attempts at installing (from the working directory, and from the root) resulted in this warning:
    npm WARN optional dep failed, continuing fsevents@0.3.6
I don't know if that's a problem or not.  It's only a warning, not listed as a fatal error (or even an error, for that matter).  I don't know what "optional dep" means.  Is it short for deprecation?  

I can re-try installing it from the working directory again. Film at 11 :)

**Comment by notewrangler** [comment]
Okay, here's what happened.  When I invoked 
    sudo su - 
I was in the project directory. the cursor changed to 
    ~#
typing 
    ls
gave me 
    node_modules, 
which was a subdirectory under the project directory.  I neglected to notice that the views folder was absent.  
typing cd .. relocated me to the root, where I could see I was nowhere near the project directory.  Navigated to there, installed express-react-views (and node-jsx, which threw the next error) and it worked.  

Thanks, that answers two questions.  I guess all that code DOES need to be in the project directory?
